### ISSUE TITLE
Polygon.fromWKT throws RangeError

### ISSUE THREAD

**Issue by RMZeroFour** [divvitco/geodart#37]
Hello. I believe there's a problem with the `Polygon.fromWKT` method.

Going through the Wikipedia page for Well-Known Text, specifically the [geometries section](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry#Geometric_objects), shows an example of the Polygon geometry as `POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))`. However, the `Polygon.fromWKT` method seems to crash when trying to parse this.

Upon reviewing the implementation, I identified two potential issues:

1. The method splits on ( and ), but a Polygon geometry uses (( and )).
2. The map lambda calls `Coordinate.fromWKT`, but it throws an error due to the spaces after the commas.

Here's a code sample for your convenience:
```dart

import 'package:geodart/geometries.dart';

main() {
  final wkt = 'POLYGON ((-64.8 32.3, -65.5 18.3, -80.3 25.2, -64.8 32.3))';
  
  // uncomment to see the error
  //print(Polygon.fromWKT(wkt));
  
  print(fixedPolygonFromWKT(wkt));
}

Polygon fixedPolygonFromWKT(String wkt) {
  // (( instead of ( and )) instead of )
  final coordinates = wkt.split('((')[1].split('))')[0].split(',');
  // c.trim() instead of c, as the spaces after commas seem to cause problems as well
  return Polygon(
      [LinearRing(coordinates.map((c) => Coordinate.fromWKT(c.trim())).toList())]);
}

```
The `fixedPolygonFromWKT` function should be a drop-in replacement for `Polygon.fromWKT` to address these issues. However, there may be edge cases that I missed, so it would be great if you could review the specification as well.

Thank you for your package, and have a nice day!

**Comment by RMZeroFour** [comment]
After some more testing, I believe the following code snippet implements Polygon parsing better, as it supports Polygons with multiple Rings as well.

```dart
import 'package:geodart/geometries.dart';

main() {
  //print(Polygon.fromWKT(
  //  'POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))'));
  print(fixedPolygonFromWKT(
      'POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))'));

  //print(Polygon.fromWKT(
  //  'POLYGON((30 10, 40 40, 20 40, 10 20, 30 10), (20 30, 35 35, 30 20, 20 30))'));
  print(fixedPolygonFromWKT(
      'POLYGON((30 10, 40 40, 20 40, 10 20, 30 10), (20 30, 35 35, 30 20, 20 30))'));
}

Polygon fixedPolygonFromWKT(String wkt) {
  wkt = wkt.substring(wkt.indexOf('(') + 1, wkt.lastIndexOf(')')).trim();

  final rings = [];
  for (int i = 0; i < wkt.length; i++) {
    if (wkt[i] == '(') {
      final end = wkt.indexOf(')', i);
      rings.add(wkt.substring(i + 1, end));
      i = end + 1;
    }
  }

  return Polygon(rings
      .map((ring) => LinearRing(
            ring
                .split(',')
                .map<Coordinate>((c) => Coordinate.fromWKT(c.trim()))
                .toList(),
          ))
      .toList());
}

```

**Comment by ntatko** [comment]
I like this, it's fantastically more clear than what's going on now. I'll get a PR up this evening. I'll also include a similar strategy for multipolygons - that probably will be slightly more complex, but it shouldn't be too bad. I'll hit you up for an approval.

**Comment by ntatko** [comment]
Sorry, it's a little later than I intended - wanna take a look?

**Comment by RMZeroFour** [comment]
No problem!

**Comment by ntatko** [comment]
Shipped in [0.3.6](https://pub.dev/packages/geodart/versions/0.3.6)
